R/g.report.part2.R

```diff
@@ -239,7 +239,8 @@ g.report.part2 = function(metadatadir = c(), f0 = c(), f1 = c(), maxdur = 0,
       # tidy up data.frames
       SUMMARY_clean = tidyup_df(SUMMARY)
       daySUMMARY_clean = tidyup_df(daySUMMARY)
-      
+      # format calendar dates (dates are stored as iso8601, so date is in the first 10 characters )
+      daySUMMARY_clean$calendar_date = substr(daySUMMARY_clean$calendar_date, 1, 10)
```

Comment:
I fixed the same issue before. Please stop using `substr` to extract info from timestamps. This is bad practise because date formats can vary so it is risky to rely on the number of characters. Instead interpret the timestamp as a timestamp and extract the date with the normal R functions for timestamp handling:

```
> time = Sys.time()
> as.Date(time)
[1] "2024-09-23"
> format(as.Date(time), "%d")
[1] "23"
> format(time, "%d/%m/%Y")
[1] "23/09/2024"
```

Reply:
Yes, this is what I first thought, but for converting the calendar dates to dates we would need the function `iso8601chartime2POSIX`, and then apply the desired format. To use the `iso8601chartime2POSIX`, we would need access to the desiredtz parameter, which is not available in `g.report.part2` and that's why I implemented this. But you are right, it might be problematic in the future. So, are you ok about including `params_general` as new input parameter to `g.report.part2`?

Reply:
If it is the only parameter from params_general then I think it is neater to only pass on desiredtz, so desiredtz would be a new argument to g.report.part2